Tests/XcodeprojTests/GenerateXcodeprojTests.swift

```diff
@@ -154,7 +154,7 @@ class GenerateXcodeprojTests: XCTestCase {
             XCTAssertNoDiagnostics(observability.diagnostics)
 
             _ = try xcodeProject(
-                xcodeprojPath: AbsolutePath.root.appending(component: "xcodeproj"),
+                xcodeprojPath: AbsolutePath(localFileSystem.currentWorkingDirectory!.components.first!).appending(component: "xcodeproj"),
```

Comment:
since we are using the local file system, should we change all operations to use directories inside the temp dir (`dstdir`)? otherwise this will have side effects on he user machine

Reply:
p.s. I realize this was there before the PR, but we might as well fix that together

Reply:
I think that it should be fine as is - we do clean up; but, I think that localizing the storage of course is better.  Does it make sense to do that as a follow up so that we can scrub through any other instances?

Reply:
oh, where do we clean up?

fwiw `testGenerateXcodeprojWithoutGitRepo` seems like it is doing something a better (uses the temp dir) to this end, so maybe we can copy that pattern?

Reply:
`testWithTemporaryDirectory ` has a `defer`ed cleanup embedded in it.  However, in the case of a crash or something, it would still leave content behind.  I'm not against the further clean up work though - I think that it is better to do that as leaving stuff in PWD is horrible.

Reply:
yea. that defer only cleans up things in the temp dir provided, not all side effects, so we need to make sure all files we create as part of the test are contained in the temp dir